src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php

```diff
@@ -89,132 +99,155 @@ public function handle(AddProductToOrderCommand $command)
     {
         $order = $this->getOrderObject($command->getOrderId());
 
-        if ($this->context->cart !== null) {
-            $oldCartRules = $this->context->cart->getCartRules();
-        } else {
-            $oldCartRules = [];
-        }
-
-        $this->assertOrderWasNotShipped($order);
+        $this->contextStateManager
+            ->setCurrency(new Currency($order->id_currency))
+            ->setCustomer(new Customer($order->id_customer));
 
-        $product = $this->getProductObject($command->getProductId(), (int) $order->id_lang);
-        $combination = $this->getCombination($command->getCombinationId());
+        try {
+            $this->assertOrderWasNotShipped($order);
 
-        $this->checkProductInStock($command->getProductId()->getValue(), $command->getCombinationId(), $command->getProductQuantity());
+            $oldCartRules = (new Cart($order->id_cart))->getCartRules();
```

Comment:
Only a suggestion if you feel like it, this could be done right after the call to `createNewOrEditExistingCart` L116:
`$oldCartRules = $cart->getCartRules();`

Reply:
:+1: